Sources/StreamChatUI/ChatMessageList/MessageComposer/ChatMessageComposerImageAttachmentsView.swift

```diff
@@ -55,39 +50,43 @@ open class _ChatMessageComposerImageAttachmentsView<ExtraData: ExtraDataTypes>:
         collectionView.dataSource = self
         collectionView.delegate = self
         collectionView.register(
-            _ChatMessageComposerImageAttachmentCollectionViewCell<ExtraData>.self,
-            forCellWithReuseIdentifier: _ChatMessageComposerImageAttachmentCollectionViewCell<ExtraData>.reuseId
+            uiConfig.messageComposer.imageAttachmentCollectionViewCell.self,
+            forCellWithReuseIdentifier: uiConfig.messageComposer.imageAttachmentCollectionViewCell.reuseId
         )
     }
     
     override open func setUpLayout() {
-        collectionView.translatesAutoresizingMaskIntoConstraints = false
-        embed(collectionView)
-        
+        flowLayout.scrollDirection = .horizontal
         flowLayout.itemSize = imagePreviewItemSize
         flowLayout.sectionInset = layoutMargins
+
+        embed(collectionView, insets: .init(top: layoutMargins.top, leading: 0, bottom: layoutMargins.bottom, trailing: 0))
+        collectionView.heightAnchor.pin(equalToConstant: imagePreviewItemSize.height).isActive = true
```

Comment:
I think this is not needed.

Reply:
Actually it is needed, it resizes the whole collectionView according to the height of `itemSize`... Probably should be overwritten to something else...

Reply:
What I think Vojta means is that the embed of the collection view is enough. And the height of the view should be defined by the super view, right ? @DominikBucher12 @VojtaStavik

Reply:
Correct @nuno-vieira 👍 Basically, this component should just accept whatever size its superview suggests.